pageserver/src/tenant.rs

```diff
@@ -271,10 +271,7 @@ impl UninitializedTimeline<'_> {
             .await
             .context("Failed to flush after basebackup import")?;
 
-        // Initialize without loading the layer map. We started with an empty layer map, and already
```

Comment:
May be leave this fragment: it actually fixes the problem with layer duplicates while restoring fro backup.

Reply:
I was asking for this earlier in this thread: https://github.com/neondatabase/neon/pull/4104#discussion_r1180303193 -- So far it seems most were for:
1. reverting all
2. include this in separate PR *after release*

Reply:
Leaving this out now would put in a delay of 20-30min for the release PR, which I am inclined not to do. You can veto this on the release PR, and create a new PR for this in the meantime. I hope that is ok.

Reply:
Ok. lets revert all

Reply:
FYI: https://github.com/neondatabase/neon/pull/4111 brings these back